compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -5683,13 +5690,17 @@ private BType checkInvocationArgs(BLangInvocation iExpr, List<BType> paramTypes,
                     BLangExpression restArg = iExpr.restArgs.get(j);
                     BType memType = j < tupleMemCount ? tupleMemberTypes.get(j) : tupleRestType;
                     checkTypeParamExpr(restArg, this.env, memType, true);
+                    if (restType != symTable.semanticError && this.resultType == symTable.semanticError) {
+                        restType = this.resultType;
+                    }
                 }
             }
         }
 
         BType retType = typeParamAnalyzer.getReturnTypeParams(env, bInvokableType.getReturnType());
-        if (Symbols.isFlagOn(invokableSymbol.flags, Flags.NATIVE)
-                && Symbols.isFlagOn(retType.flags, Flags.PARAMETERIZED)) {
+        if (restType != symTable.semanticError &&
```

Comment:
Why do we need this check? If there's an issue in the rest type, shouldn't the type builder handle it? 🤔

Reply:
If there is an issue with the rest arg that would mean that the arguments passed are incorrect, right? IINM, the resolved type builder assumes that the arguments are provided correctly. For non rest args we seem to be returning at L5589 itself if there is an error.

Anyway given that the return type depends on the arguments IMO it is OK for the resolved type builder to expect a correct function/method call expression since even a simple change to fix the arguments can affect the return type.